includes/ConsoleTasks/OldRequestCleanupTask.php

```diff
@@ -8,24 +9,154 @@
 
 namespace Waca\ConsoleTasks;
 
+use Waca\PdoDatabase;
 use Waca\Tasks\ConsoleTaskBase;
 
 class OldRequestCleanupTask extends ConsoleTaskBase
 {
     public function execute()
     {
-        $statement = $this->getDatabase()->prepare(<<<SQL
+        $database = $this->getDatabase();
+        $expiryTime = [':expiry' => $this->getSiteConfiguration()->getEmailConfirmationExpiryDays()];
+
+        // start by fetching the number of unconfirmed requests which have expired
+        $eligibleRecords = $this->getExpiredCount($database, $expiryTime);
+
+        // fetch the number of unconfirmed requests which have expired and which have no FK constraints which would
+        // otherwise prevent their deletion
+        $eligibleUnconstrainedRecords = $this->getExpiredUnconstrainedCount($database, $expiryTime);
+
+        // Delete any requester comments for expired requests
+        $requesterCommentDelete = <<<SQL
+            DELETE FROM comment
+            WHERE 1 = 1 
+                -- only requester comments
+                AND comment.visibility = 'requester'
+                -- where the following record exists
+                AND exists(
+                    SELECT 1 FROM request r
+                    WHERE 1 = 1
+                        -- a request matching the currently-checked comment
+                        AND comment.request = r.id
+                        -- and the request is expired
+                        AND r.date < DATE_SUB(CURRENT_TIMESTAMP(), INTERVAL :expiry DAY)
+                        -- no confirmed email address
+                        AND r.emailconfirm <> 'Confirmed'
+                        -- not already marked as stale
+                        AND r.emailconfirm <> 'Stale'
+                        -- email confirmation was requested
+                        AND r.emailconfirm <> ''
```

Comment:
Something seems fishy about this section.
The and statements would seemingly nullify finding anything.
The first comment also doesn't seem to match the code.
It could be that I'm not firing on all cylinders though.

Reply:
There's four possible values for this column.

* An empty string, which indicates that email confirmation is disabled and thus the request shows immediately in the UI
* `Confirmed` which indicates the requester has clicked the confirm link in their email
* `Stale` (which is newly added by this script)
* A hex string indicating the request is still pending email-confirmation

The AND statements exclude the three static values, leaving only the hex string confirmation hash.

The empty string and `Confirmed` both show in the UI, so will be dealt with and set to closed by humans. The `Stale` value is only set at the same time this script closes a request. Once the request is closed, clearing data is picked up by the other private data stripping script, rather than being completely deleted by this script.